File: frontend/types/group.ts
// 群组信息
export interface GroupInfo {
  group_id: string
  name: string
  description?: string
  creator_id: number
  creator_name: string
  created_at: string
  last_active_at: string
  latitude: number
  longitude: number
  member_count: number
  distance: number
  location_name: string
  is_password_required: boolean
  is_member: boolean
  user_role: string
}

// 创建群组请求
export interface CreateGroupRequest {
  name: string
  description?: string
  password?: string
  latitude: number
  longitude: number
  location_name: string
}

// 创建群组响应
export interface CreateGroupResponse {
  group_id: string
}

// 加入群组请求
export interface JoinGroupRequest {
  group_id: string
  password?: string
}

// 加入群组响应
export interface JoinGroupResponse {
  success: boolean
  role: string
}

// 离开群组请求
export interface LeaveGroupRequest {
  group_id: string
}

// 离开群组响应
export interface LeaveGroupResponse {
  success: boolean
}

// 获取群组成员请求
export interface GroupMembersRequest {
  group_id: string
  cursor?: number
  limit?: number
}

// 群组成员信息
export interface GroupMember {
  user_id: number
  nickname: string
  last_active: string
  role: string
  join_time: string
}

// 获取群组成员响应
export interface GroupMembersResponse {
  members: GroupMember[]
  has_more: boolean
  next_cursor?: number
}

// 按位置搜索群组请求
export interface SearchGroupByLocationRequest {
  latitude: number
  longitude: number
  radius: number
  cursor?: number
  limit?: number
}

// 按名称搜索群组请求
export interface SearchGroupByNameRequest {
  keyword: string
  cursor?: number
  limit?: number
} 

File: backend/src/models/api/group.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

// API请求模型
#[derive(Debug, Serialize, Deserialize)]
pub struct CreateGroupRequest {
    pub name: String,
    pub description: Option<String>,
    pub password: Option<String>,
    pub latitude: f64,
    pub longitude: f64,
    pub location_name: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateGroupResponse {
    pub group_id: Uuid,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct JoinGroupRequest {
    pub group_id: Uuid,
    pub password: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct JoinGroupResponse {
    pub success: bool,
    pub role: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct LeaveGroupRequest {
    pub group_id: Uuid,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct LeaveGroupResponse {
    pub success: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct KickMemberRequest {
    pub group_id: Uuid,
    pub user_id: i64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct KickMemberResponse {
    pub success: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateGroupRequest {
    pub group_id: Uuid,
    pub name: Option<String>,
    pub description: Option<String>,
    pub avatar_url: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateGroupResponse {
    pub success: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetGroupRequest {
    pub group_id: Uuid,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GroupInfo {
    pub group_id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub owner_id: i64,
    pub creator_name: String,
    pub created_at: DateTime<Utc>,
    pub last_active_at: DateTime<Utc>,
    pub latitude: f64,
    pub longitude: f64,
    pub member_count: i64,
    pub distance: f64,
    pub location_name: String,
    pub is_password_required: bool,
    pub is_member: bool,
    pub user_role: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetGroupResponse {
    pub group: GroupInfo,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GroupMembersRequest {
    pub group_id: Uuid,
    pub cursor: Option<i64>,
    pub limit: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GroupMemberInfo {
    pub user_id: i64,
    pub nickname: String,
    pub last_active: String,
    pub role: String,
    pub join_time: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GroupMembersResponse {
    pub members: Vec<GroupMemberInfo>,
    pub has_more: bool,
    pub next_cursor: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchGroupByLocationRequest {
    pub latitude: f64,
    pub longitude: f64,
    pub radius: f64,
    pub cursor: Option<i64>,
    pub limit: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchGroupByNameRequest {
    pub keyword: String,
    pub cursor: Option<i64>,
    pub limit: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchGroupByIdRequest {
    pub group_id: Uuid,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchGroupByLocationResponse {
    pub groups: Vec<GroupInfo>,
    pub has_more: bool,
    pub next_cursor: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SearchGroupByNameResponse {
    pub groups: Vec<GroupInfo>,
    pub has_more: bool,
    pub next_cursor: Option<i64>,
}

pub type SearchGroupByIdResponse = GroupInfo;

#[derive(Debug, Serialize, Deserialize)]
pub struct TransferOwnershipRequest {
    pub group_id: Uuid,
    pub new_owner_id: i64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TransferOwnershipResponse {
    pub success: bool,
}
